### PR TITLE
Allow to split non-leaf frames

### PR DESCRIPTION
**Pull request by t-wissmann**
This extends the 'split' command by an optional argument containing a
frame index. The FrameTree::splitCommand() was quite specific to
FrameLeafs, so I had to change the type of 'frame' and tried to keep the
meaning of the other variables as good as possible for the case where
'frame' is a FrameSplit.

The completion of the 'split' command is still defined in command.cpp.
Migrating this is in the scope of another command.

This implements #906.

### CODE REVIEW

**Comment by t-wissmann** [comment]
This PR will have file conflits with #974 because both mark the `FrameTree::replaceNode` function as public.

File: NEWS [hunk 1]
```diff
@@ -9,6 +9,7 @@ Current git version
   * Fix handling of delta -1 in 'focus_monitor' and 'cycle_monitor'
   * Fixed precision decimals in the layout tree (more reliable in- and output
     of fractions in frame splits)
+  * New optional argument 'split' for splitting non-leaf frames
```
Comment (blarz) [inline NEWS:12]: "New optional argument for the 'split' command" maybe?

File: doc/herbstluftwm.txt [hunk 2]
```diff
@@ -398,6 +399,7 @@ split 'ALIGN' ['FRACTION']::
         * split horiz 0.3
         * split vertical 0.5
         * split h
+        * split top 0.2 ''   (splits the root frame)
```
Comment (blarz) [inline doc/herbstluftwm.txt:402]: So an empty frame index means root frame? Is this "known"? ;)

File: NEWS (hunk 1 above)
Comment (t-wissmann) [reply NEWS]: Yes; fixed!

File: doc/herbstluftwm.txt (hunk 2 above)
Comment (t-wissmann) [reply doc/herbstluftwm.txt:402]: Yes, it's documented in the section "frame index" (I've added a reference)

Comment (blarz) [reply doc/herbstluftwm.txt:402]: Awesome, thank you!

File: NEWS (hunk 1 above)
Comment (blarz) [reply NEWS]: Thanks!

**Pull request merged**